posthog/warehouse/api/managed_viewset.py

```diff
@@ -0,0 +1,127 @@
+import structlog
+from rest_framework import serializers, status, viewsets
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from posthog.api.routing import TeamAndOrgViewSetMixin
+from posthog.warehouse.models import ManagedViewSet
+
+logger = structlog.get_logger(__name__)
+
+
+class ManagedViewSetSerializer(serializers.Serializer):
+    enabled = serializers.BooleanField(required=True)
+
+
+class ManagedViewSetViewSet(TeamAndOrgViewSetMixin, viewsets.ViewSet):
+    scope_object = "INTERNAL"
+    lookup_field = "kind"
+    lookup_url_kwarg = "kind"
+
+    def list(self, request: Request, *args, **kwargs) -> Response:
+        """
+        Get all views associated with managed viewsets.
+        GET /api/environments/{team_id}/managed_viewsets/
+        """
+        from posthog.warehouse.models import DataWarehouseSavedQuery
+
+        managed_viewsets = ManagedViewSet.objects.filter(team_id=self.team_id)
+
+        views = (
+            DataWarehouseSavedQuery.objects.filter(
+                team_id=self.team_id,
+                managed_viewset__in=managed_viewsets,
```

Comment:
Can you not do `managed_viewset__isnull=False` instead of doing a lookup on `ManagedViewSet` first and doing a `in` clause?